graylog2-server/src/main/java/org/graylog2/buffers/Buffers.java

```diff
@@ -22,47 +22,76 @@
 import com.github.rholder.retry.StopStrategies;
 import com.github.rholder.retry.WaitStrategies;
 import com.google.common.base.Predicates;
+import org.graylog2.plugin.buffers.EventBuffer;
+import org.graylog2.plugin.buffers.InputBuffer;
 import org.graylog2.shared.buffers.ProcessBuffer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.inject.Inject;
+import java.util.EnumSet;
+import java.util.LinkedHashMap;
 import java.util.Locale;
-import java.util.concurrent.Callable;
+import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.graylog2.buffers.Buffers.Type.INPUT;
+import static org.graylog2.buffers.Buffers.Type.OUTPUT;
+import static org.graylog2.buffers.Buffers.Type.PROCESS;
 
 public class Buffers {
+    public enum Type {
+        INPUT,
+        PROCESS,
+        OUTPUT,
+    }
+
     private static final Logger LOG = LoggerFactory.getLogger(Buffers.class);
     private static final long DEFAULT_MAX_WAIT = 30L;
 
+    private final InputBuffer inputBuffer;
     private final ProcessBuffer processBuffer;
     private final OutputBuffer outputBuffer;
 
     @Inject
-    public Buffers(final ProcessBuffer processBuffer, final OutputBuffer outputBuffer) {
+    public Buffers(InputBuffer inputBuffer, final ProcessBuffer processBuffer, final OutputBuffer outputBuffer) {
+        this.inputBuffer = inputBuffer;
         this.processBuffer = processBuffer;
         this.outputBuffer = outputBuffer;
     }
 
+    /**
+     * @deprecated Use {@link #waitForEmptyBuffers(EnumSet)} instead
+     */
+    @Deprecated
     public void waitForEmptyBuffers() {
         waitForEmptyBuffers(DEFAULT_MAX_WAIT, TimeUnit.SECONDS);
     }
 
+    public void waitForEmptyBuffers(EnumSet<Type> bufferTypes) {
```

Comment:
why would we need this method? `bufferTypes` is not used

Reply:
Thanks for noticing. The method was meant as a replacement for `#waitForEmptyBuffers()`. Fixed the unused parameter and started to actually use the method.